=== modified file 'src/modbus.c'
--- src/modbus.c	2007-11-02 18:22:38 +0000
+++ src/modbus.c	2008-02-10 08:46:19 +0000
@@ -1147,10 +1147,11 @@
 }
 
 /* Initialises the modbus_param_t structure for TCP */
-void modbus_init_tcp(modbus_param_t *mb_param, char *ip)
+void modbus_init_tcp(modbus_param_t *mb_param, char *ip, uint16_t port)
 {
         memset(mb_param, 0, sizeof(modbus_param_t));
         strncpy(mb_param->ip, ip, sizeof(char)*16);
+        mb_param->port = port;
         mb_param->type_com = TCP;
         mb_param->header_length = HEADER_LENGTH_TCP;
         mb_param->checksum_size = CHECKSUM_SIZE_TCP;
@@ -1418,7 +1419,7 @@
         struct sockaddr_in addr;
 
         addr.sin_family = AF_INET;
-        addr.sin_port = htons(MODBUS_TCP_PORT);
+        addr.sin_port = htons(mb_param->port);
         addr.sin_addr.s_addr = inet_addr(mb_param->ip);
 
         mb_param->fd = socket(AF_INET, SOCK_STREAM, 0);
@@ -1541,7 +1542,7 @@
         addr.sin_family = AF_INET;
         /* The modbus port is < 1024
            This program must be made setuid root. */
-        addr.sin_port = htons(MODBUS_TCP_PORT);
+        addr.sin_port = htons(mb_param->port);
         addr.sin_addr.s_addr = INADDR_ANY;
         memset(&(addr.sin_zero), '\0', 8);
 

=== modified file 'src/modbus.h'
--- src/modbus.h	2007-11-12 23:02:49 +0000
+++ src/modbus.h	2008-02-10 08:46:57 +0000
@@ -25,8 +25,6 @@
 #include <termios.h>
 #include <arpa/inet.h>
 
-#define MODBUS_TCP_PORT      502
-
 #define HEADER_LENGTH_RTU         0
 #define PRESET_QUERY_SIZE_RTU     6
 #define PRESET_RESPONSE_SIZE_RTU  2
@@ -132,6 +130,8 @@
         int debug;
         /* IP address */
         char ip[16];
+        /* Tcp port */
+        uint16_t port;
         /* Header length used for offset */
         int header_length;
         /* Checksum size RTU = 2 and TCP = 0 */
@@ -205,10 +205,11 @@
 void modbus_init_rtu(modbus_param_t *mb_param, char *device,
                      int baud, char *parity, int data_bit,
                      int stop_bit);
+                     
 /* Initialises a parameters structure for TCP
-   - ip : "192.168.0.5" */
-void modbus_init_tcp(modbus_param_t *mb_param, char *ip_address);
-
+   - ip : "192.168.0.5" 
+   - port : 502 */
+void modbus_init_tcp(modbus_param_t *mb_param, char *ip_address, uint16_t port);
 
 /* Sets up a serial port for RTU communications to modbus or a TCP
    connexion */

=== modified file 'src/test-modbus-master.c'
--- src/test-modbus-master.c	2007-11-02 16:31:24 +0000
+++ src/test-modbus-master.c	2008-02-10 08:48:52 +0000
@@ -47,7 +47,7 @@
 /*      modbus_init_rtu(&mb_param, "/dev/ttyS0", 19200, "none", 8, 1); */
 
         /* TCP */
-        modbus_init_tcp(&mb_param, "169.254.7.104");
+        modbus_init_tcp(&mb_param, "127.0.0.1", 502);
         modbus_set_debug(&mb_param, TRUE);
       
         modbus_connect(&mb_param);

=== modified file 'src/test-modbus-slave.c'
--- src/test-modbus-slave.c	2007-11-02 16:31:24 +0000
+++ src/test-modbus-slave.c	2008-02-10 08:49:20 +0000
@@ -38,7 +38,7 @@
         int ret;
         int i;
 
-        modbus_init_tcp(&mb_param, "127.0.0.1");
+        modbus_init_tcp(&mb_param, "127.0.0.1", 502);
         modbus_set_debug(&mb_param, TRUE);
 
         modbus_mapping_new(&mb_mapping,

